pkg/tool/default.go

```diff
@@ -0,0 +1,106 @@
+package tool
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/openshift/backplane-tools/pkg/source/github"
+	"github.com/openshift/backplane-tools/pkg/utils"
+)
+
+type DefaultTool struct {
+	Tool
+	name       string
+	binaryName string // must be retrieved by BinaryName() func
```

Comment:
Can we make this a `*string` to show that this is optional? We could also remove the comment here, as it might just be confusing. It's clear that if this is private it has to be fetched via a getter.

Reply:
binaryName is not really optional, it is just that by default it should be set to name but in some cases it should b set to other values.
In order to do so, we ask to always use BinaryName() to retireve it (not to say that it is private, as even inside the object we should use the getter, which is the goal of the comment).
There might be other way to achieve this, but as in golang we cannot override funct definition in subclass, we need to use workarround like that.

Indeed an other approach would be to define binaryName .string and ask BinaryName() to return binaryName if not nil or name otherwise...
Would you prefer this and want me to change in that way ?

Reply:
Not to be overly pedantic, since I'm probably more guilty of conflating the two than anyone else, but _technically_ this should be called `executableName` rather than `binaryName`. 

There's no real requirement that the tools we install be compiled binaries; in fact, the awscli tool wraps its executable (which I think is python?) in a bash wrapper :slightly_smiling_face:

Reply:
I did not used the *string, as it makes it more complicated to pass const string as parameters, but I made it clear that empty string (default in struct) means we will use name for binaryName

Reply:
binaryName renamed executableName